presto-main/src/main/java/com/facebook/presto/server/remotetask/WorkerTaskStatusFetcher.java

```diff
@@ -0,0 +1,284 @@
+import com.facebook.presto.server.smile.SmileCodec;
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.Duration;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_CURRENT_STATE;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_MAX_WAIT;
+import static com.facebook.presto.server.RequestHelpers.setContentTypeHeaders;
+import static com.facebook.presto.server.smile.AdaptingJsonResponseHandler.createAdaptingJsonResponseHandler;
+import static com.facebook.presto.server.smile.FullSmileResponseHandler.createFullSmileResponseHandler;
+import static com.facebook.presto.server.smile.JsonCodecWrapper.unwrapJsonCodec;
+import static com.facebook.presto.spi.StandardErrorCode.REMOTE_TASK_MISMATCH;
+import static com.facebook.presto.util.Failures.REMOTE_TASK_MISMATCH_ERROR;
+import static io.airlift.units.Duration.nanosSince;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+class WorkerTaskStatusFetcher
+        implements SimpleHttpResponseCallback<List<TaskStatus>>
+{
+    private final ConcurrentHashMap<TaskId, ContinuousBatchTaskStatusFetcher.Task> idTaskMap;
+
+    private static final Logger log = Logger.get(ContinuousTaskStatusFetcher.class);
+
+    private final TaskId taskId; // TODO: Remove after removing references
+    private final StateMachine<TaskStatus> taskStatus; // TODO: Remove after removing references
+
+    private final Consumer<Throwable> onFail;
+    private final Codec<List<TaskStatus>> taskListStatusCodec;
+    private final Duration refreshMaxWait;
+    private final Executor executor;
+    private final HttpClient httpClient;
+    private final RequestErrorTracker errorTracker;
+    private final RemoteTaskStats stats;
+    private final boolean isBinaryTransportEnabled;
+
+    private final AtomicLong currentRequestStartNanos = new AtomicLong();
+
+    @GuardedBy("this")
+    private boolean running;
+
+    @GuardedBy("this")
+    private ListenableFuture<BaseResponse<TaskStatus>> future;
+
+    public WorkerTaskStatusFetcher() {
+        this(null, null, null, null, null, null, null, null, null, null, false);
+    }
+
+    // TODO: We don't need all the parameters here and the logic should be in CBTSF
+    public WorkerTaskStatusFetcher(
```

Comment:
we could just remove this constructor right?

Reply:
Removed other constructor and left necessary params in this one.